pkg/prometheus/promcfg.go

```diff
@@ -59,10 +60,11 @@ type ConfigGenerator struct {
 	notCompatible          bool
 	prom                   monitoringv1.PrometheusInterface
 	endpointSliceSupported bool
+	ctx                    context.Context
```

Comment:
Its preferred to pass context as first argument to function than having it part of struct  
https://go.dev/blog/context-and-structs

Reply:
Cool! Thanks for sharing @slashpai I didn't know this practice.
Can you take a look again?